generated/src/aws-cpp-sdk-s3/source/S3ErrorMarshaller.cpp

```diff
@@ -127,27 +127,35 @@ AWSError<Aws::Client::CoreErrors> S3ErrorMarshaller::Marshall(const Aws::Http::H
   XmlDocument doc = XmlDocument::CreateFromXmlStream(body);
   body.seekg(readPointer);
   Aws::String bodyError;
+  Aws::String requestId;
 
   if (doc.WasParseSuccessful() &&
       !doc.GetRootElement().IsNull() && 
       doc.GetRootElement().GetName() == Aws::String("Error")) 
   {        
       //check if the first node fetched has no children
       auto messageNode = doc.GetRootElement().FirstChild("Message") ;
-      if(!messageNode.IsNull() && !messageNode.HasChildren())
+      if(!messageNode.IsNull())
       {
           message = messageNode.GetText();
       }
       auto codeNode = doc.GetRootElement().FirstChild("Code") ;
-      if(!codeNode.IsNull() && !codeNode.HasChildren())
+      if(!codeNode.IsNull())
```

Comment:
same question for the error code, do we need to recurse down the tree to find first valid (non-empty) text

Reply:
same response as other question, this wasn't validating leaf, leaf has a empty child